### ISSUE TITLE
Why convert to cpu during StAS

### ISSUE THREAD

**Issue by Amandaynzhou** [malllabiisc/ASAP#1]
hello,

Thanks for the code sharing. 
Is there any reason to convert tensor to cpu during StAS?
`index_E, value_E =spspmm(index_St.cpu(), value_St.cpu(), index_B.cpu(), value_B.cpu(), kN, N, kN)`
Instead of directly compute in gpu?
`index_E, value_E = spspmm(index_St, value_St, index_B, value_B, kN, N, kN)`

When I use torch==1.4, torch geometric==1.4.3, torch-sparse==0.6.1 it will cause a NAN error. But directly calculating in GPU is correct.

**Comment by ekagra-ranjan** [comment]
Hi, 

The reason for performing StAS on cpu was that we were occasionally getting NAN error on GPU. Switching to CPU somehow solved this error (although we couldn't figure out why). It is surprising that the situation is reversed in your case. Nonetheless, if you are able to run on GPU consistently then its fine as well.

**Comment by Amandaynzhou** [comment]
Thanks for your response :)

**Comment by rishabhjoshi** [comment]
I would like to add that I was using ASAP in another code (different graphs and all), and I faced the issue of device side assert triggered / index out of bounds at the loss.backward() step when I was using the GPU variant of the snippet. When I changed back to the older version of doing .cpu() and then later .to(device), it started working. I was not getting any NaN errors.

**Comment by ekagra-ranjan** [comment]
@rishabhjoshi  Thank you for sharing your insight.

 Recently, ASAP was added to PyG ([ASAP docs](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.pool.ASAPooling)). Could you check whether using ASAP from PyG directly is still causing Nan on GPUs on your dataset? I think the problem is coming from `pytorch_sparse` but might be resolved in the PyG version due to the usage of the newly introduced `SparseTensor`.
